### ISSUE TITLE
Roslyn shows a false error for an interceptor attribute when pathmap is used

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#70138]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.8.0 Preview 1.0
VisualStudio.17.Preview/17.8.0-pre.1.0+34004.107
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Compile the following project:
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <LangVersion>preview</LangVersion>
    <Features>InterceptorsPreview</Features>
    <PathMap>C:\Work\TestNet7App\ConsoleApp1=/_/</PathMap>
  </PropertyGroup>
</Project>
```
with a single file:
```cs
using System;
using System.Runtime.CompilerServices;

var c = new C();
c.Intercepted();

static class InterceptorMethods
{
  [InterceptsLocation("/_/Program.cs", line: 5, character: 3)]
  public static void Intercept(this C c) { }
}

namespace System.Runtime.CompilerServices
{
  [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
  public class InterceptsLocationAttribute : System.Attribute
  {
    public InterceptsLocationAttribute(string path, int line, int character) { }
  }
}

class C
{
  public void Intercepted() { }
}
```

**Expected Behavior**:
No errors are reported in code editor. The project compiles and runs without a problem

**Actual Behavior**:
An error highlighting is shown in the editor and an error is reported in error list claiming that the path `/_/Program.cs` doesn't exist. It does, although it comes from `<PathMap>`

![image](https://github.com/dotnet/roslyn/assets/5566206/9765946a-5dde-4143-8c44-9a7ee46efb93)

**Notes**:
It looks like the design time build ignores the pathmap property.
The current behavior makes it impossible to "fix" the error when pathmapping is required.
Either you always see the error always reported in the error view, or the project won't compile properly because at the build time the path will be different.

**Comment by CyrusNajmabadi** [comment]
Is this an IDE-only error?  Specifically, is it possible to successfully compile from the command line, but you're just getting spurious IDE squiggles?

**Comment by TessenR** [comment]
This is an editor and error list -only error. 
I.e. you don't even have to switch to the command line. You can just build and run the solution directly from VS. An error in the editor and the error list doesn't prevent it from being built and run. Check the output window on the provided screenshot - the build command works without any issues.
